tests/cli.rs

```diff
@@ -0,0 +1,225 @@
+#![cfg(any(target_os = "windows", target_os = "macos", target_os = "linux"))]
+use std::env;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::{Path, PathBuf};
+use std::process::{Child, ChildStderr, Command, Stdio};
+
+use anyhow::{Context, Result};
+use tempfile::tempdir;
+
+#[test]
+fn cli_provide_one_file() -> Result<()> {
+    // provide a path to a file, do not pipe from stdin, do not pipe to stdout
+    test_provide_get_loop(&PathBuf::from("transfer").join("foo.bin"), false, false)
+}
+
+#[test]
+fn cli_provide_folder() -> Result<()> {
+    // provide a path to a folder, do not pipe from stdin, do not pipe to stdout
+    test_provide_get_loop(&PathBuf::from("transfer"), false, false)
+}
+
+#[test]
+fn cli_provide_from_stdin_to_stdout() -> Result<()> {
+    // provide a file, pipe content to the provider's stdin, pipe content to the getter's stdout
+    test_provide_get_loop(&PathBuf::from("transfer").join("foo.bin"), true, true)
+}
+
+// Test the provide and get loop for success, stderr output, and file contents.
+//
+// Can optionally pipe the given `path` content to the provider from stdin & can optionally save the output to an `out` path.
+//
+// Runs the provider as a child process that stays alive until the getter has completed. Then
+// checks the output of the "provide" and "get" processes against expected regex output. Finally,
+// test the content fetched from the "get" process is the same as the "provided" content.
+fn test_provide_get_loop(path: &Path, use_stdin: bool, use_stdout: bool) -> Result<()> {
```

Comment:
it would be much more readable to make the `bool`s an enum like this

```rust
enum Output {
  Path,
  Stdout,
}

enum Input {
  Path,
  Stdin
}

```

Reply:
ty this was bothering me & I didn't understand the clearest way to fix it! 🔥